src/main/java/org/jabref/gui/maintable/SmartConstrainedResizePolicy.java

```diff
@@ -22,75 +23,158 @@
+                        for (TableColumnBase<?, ?> col : visibleLeafColumns) {
+                            Double share = expansionShare.get(col);
+                            Double newSize;
+                            if (col.equals(prop.getColumn())) {
+                                // The column resized by the user gets the delta;
+                                newSize = newWidth;
+                            } else {
+                                // The columns not explicitly resized by the user get the negative delta
+                                newSize = col.getWidth() - share * deltaToApply;
+                            }
+                            newSize = Math.min(newSize, col.getMaxWidth());
+                            col.setPrefWidth(newSize);
+                        }
+                    });
+            return true;
+        }
+
+        LOGGER.debug("Window smaller than content");
+
+        determineNewWidth(userChosenColumnToResize, delta).ifPresent(newWidth ->
+                userChosenColumnToResize.setPrefWidth(newWidth));
+        return true;
     }
 
-    private Boolean constrainedResize(TableView.ResizeFeatures prop, Boolean isFirstRun, Double contentWidth, List<? extends TableColumnBase<?, ?>> visibleLeafColumns) {
-        // We have to use reflection since TableUtil is not visible to us
-        try {
-            // TODO: reflective access, should be removed
-            Class<?> clazz = Class.forName("javafx.scene.control.TableUtil");
-            Method constrainedResize = clazz.getDeclaredMethod("constrainedResize", ResizeFeaturesBase.class, Boolean.TYPE, Double.TYPE, List.class);
-            constrainedResize.setAccessible(true);
-            Object returnValue = constrainedResize.invoke(null, prop, isFirstRun, contentWidth, visibleLeafColumns);
-            return (Boolean) returnValue;
-        } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException | ClassNotFoundException e) {
-            LOGGER.error("Could not invoke constrainedResize in TableUtil", e);
-            return false;
+    /**
+     * Rearranges the widths of all columns according to their shares (proportional to their preferred widths)
+     */
+    private Boolean rearrangeColumns(TableView<?> table) {
+        Double tableWidth = getContentWidth(table);
+        List<? extends TableColumnBase<?, ?>> visibleLeafColumns = table.getVisibleLeafColumns();
+
+        Double remainingAvailableWidth = tableWidth - getSumMinWidthOfColumns(table);
+
+        for (TableColumnBase<?, ?> col : visibleLeafColumns) {
+            double share = expansionShare.get(col);
+            double newSize = col.getMinWidth() + share * remainingAvailableWidth;
+
+            // Just to make sure that we are staying under the total table width (due to rounding errors)
+            newSize -= 2;
+
+            newSize = Math.min(newSize, col.getMaxWidth());
+
+            col.setPrefWidth(newSize);
```

Comment:
I think this might lead to unused space on the right, when one of the columns has a higher preferred width than it's maximum (because you don't redistribute the difference among the other columns).

Reply:
That's why the resize method in javafx returns the unused delta:
https://github.com/openjdk/jfx/blob/53fe38bb34fc01ba298aa1a12f20c061c0cb58b2/modules/javafx.controls/src/main/java/javafx/scene/control/TableUtil.java#L363-L366